changes/2020-06-09_wrapping-key-identifiers/change.md

```diff
@@ -0,0 +1,151 @@
+| [AWS KMS Keyring](../../framework/kms-keyring.md)         |
+| [Raw AES Keyring](../../framework/raw-aes-keyring.md)     |
+| [Raw RSA Keyring](../../framework/raw-rsa-keyring.md)     |
+
+## Affected Implementations
+
+The scope of this change only affects the specification.
+Follow-up changes that define
+whether implementations expose these concepts
+MAY affect implementations.
+
+## Definitions
+
+### Conventions used in this document
+
+The key words
+"MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT",
+"SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL"
+in this document are to be interpreted as described in
+[RFC 2119](https://tools.ietf.org/html/rfc2119).
+
+## Summary
+
+We have referenced several terms throughout various specification documents
+that are never defined:
+"key namespace",
+"key name",
+"provider ID",
+and "provider info".
+This change defines each of these terms
+and describes their relationships to each other
+and the keyrings they represent.
+
+## Out of Scope
+
+- Whether, or how, each keyring exposes any of these values at runtime is out of scope.
+
+- Specific values for these terms for any specific keyring is out of scope.
+
+## Motivation
+
+Various specification documents reference
+"key namespace",
+"key name",
+"provider ID",
+and "provider info",
+but we never define what these terms mean
+or how they relate to each other.
+
+In order to ensure the specification documents accurately describe keyring behavior,
+we need to define all terms that they use.
+
+## Drawbacks
+
+This change SHOULD NOT introduce any drawbacks.
+It is identifying and describing
+terms that already exist
+and the existing relationship between them.
+
+## Security Implications
+
+This change SHOULD NOT have any security implications.
+
+## Operational Implications
+
+This change SHOULD NOT have any operational implications.
+
+## Guide-level Explanation
+
+Key namespace and key name are configuration values that determine the behavior of a keyring.
+
+Provider ID and provider info are values that identify the keyring configuration
+that can fulfill the keyring's decryption contract.
+The keyring attaches these values to the encrypted data key ciphertext.
+
+## Reference-level Explanation
+
+"Key namespace", "key name", "provider ID", and "provider info"
+are all concepts that identify a wrapping key.
+
+### key namespace
+
+A configuration value for a keyring
+that identifies the grouping or categorization
+for the wrapping keys that keyring can access.
+
+The key namespace MUST be a string value.
+
+### key name
+
+A configuration value for a keyring
+that identifies a single wrapping key
+within a key namespace.
+
+The key name MUST be a string value.
+
+### provider ID
+
+An output value returned by a keyring on encrypt
+as part of an encrypted data key structure
+that identifies the grouping or categorization
+for the keyring that can fulfill this keyring's decryption contract.
```

Comment:
```suggestion
for a keyring to fulfill this its decryption contract.
```

From #131, I'm still wondering about where
the decryption contract is.
I _think_ that the contract **is** the encrypted data key?
But then what do we say about keyrings,
that that can unwrap the encrypted data key
but chose not to?

Reply:
> But then what do we say about keyrings,
that that can unwrap the encrypted data key
but chose not to?

In that scenario I would say that is an example of a keyring being "unwilling" (as opposed to "unable") to fulfill the decryption contract.